sos/plugins/juju_controller.py

```diff
@@ -0,0 +1,193 @@
+            "Export mongodb collections as json files",
+            "",
+            False,
+        )
+    ]
+
+    def export_mongodb(self, machineid, dbpass):
+        collections = [
+            "actionnotifications",
+            "actions",
+            "annotations",
+            "applicationOfferConnections",
+            "applicationOffers",
+            "applications",
+            "bakeryStorageItems",
+            "blockdevices",
+            "charms",
+            "cleanups",
+            "cloudCredentials",
+            "cloudimagemetadata",
+            "clouds",
+            "constraints",
+            "containerRefs",
+            "controllers",
+            "controllerusers",
+            "deviceConstraints",
+            "endpointbindings",
+            "filesystems",
+            "generations",
+            "globalRefcounts",
+            "globalSettings",
+            "instanceData",
+            "ip.addresses",
+            "leaseholders",
+            "leases",
+            "linklayerdevices",
+            "linklayerdevicesrefs",
+            "machineUpgradeSeriesLocks",
+            "machineremovals",
+            "machines",
+            "managedStoredResources",
+            "meterStatus",
+            "metricsmanager",
+            "migrations",
+            "modelEntityRefs",
+            "modelUserLastConnection",
+            "models",
+            "modelusers",
+            "openedPorts",
+            "payloads",
+            "permissions",
+            "providerIDs",
+            "refcounts",
+            "relationNetworks",
+            "relations",
+            "relationscopes",
+            "remoteEntities",
+            "resources",
+            "sequence",
+            "settings",
+            "sshhostkeys",
+            "statuses",
+            "storageattachments",
+            "storageconstraints",
+            "storageinstances",
+            "storedResources",
+            "subnets",
+            "toolsmetadata",
+            "txns.prune",
+            "txns.stash",
+            "units",
+            "userLastLogin",
+            "usermodelname",
+            "users",
+            "volumes",
+        ]
+        # We don't collect txns.log, statuseshistory, metrics, txns due to size
+
+        for collection in collections:
+            filename = "mongoexport_collection_{}.json".format(collection)
+            # This will execute a mongoexport command, note it will add the
+            # password to the sos log
+            mongocmd = (
+                "mongoexport --host 127.0.0.1 --port 37017 --db juju "
+                + "--authenticationDatabase admin --ssl "
+                + "--sslAllowInvalidCertificates --username '{}' --password "
+                + "'{}' --collection {} --jsonArray"
+            )
+            self.add_cmd_output(
+                mongocmd.format(machineid, dbpass, collection),
+                suggest_filename=filename,
+            )
+
+    def get_machine_id(self):
```

Comment:
Not a great name, since `machine_id` now has a system wide meaning - consider `juju_machine_id`.